lib/rbs/prototype/rb.rb

```diff
@@ -220,16 +239,55 @@ def process(node, decls:, comments:, singleton:)
               decls << AST::Members::Alias.new(
                 new_name: new_name,
                 old_name: old_name,
-                kind: singleton ? :singleton : :instance,
+                kind: context.singleton ? :singleton : :instance,
                 annotations: [],
                 location: nil,
                 comment: comments[node.first_lineno - 1],
               )
             end
-          end
+          when :module_function
+            if args.empty?
+              context.module_function = true
+            else
+              module_func_context = context.dup.tap { |ctx| ctx.module_function = true }
+              args.each do |arg|
+                if arg && (name = literal_to_symbol(arg))
+                  i = find_def_index_by_name(decls, name)
+                  decls[i] = decls[i].update(kind: :singleton_instance)
+                elsif arg
+                  process arg, decls: decls, comments: comments, context: module_func_context
+                end
+              end
+            end
+          when :public, :private
+            accessibility = __send__(node.children[0])
+            if args.empty?
+              decls << accessibility
+            else
+              args.each do |arg|
+                if arg && (name = literal_to_symbol(arg))
+                  if i = find_def_index_by_name(decls, name)
+                    current = current_accessibility(decls, i)
+                    if current != accessibility
+                      decls.insert(i + 1, current)
+                      decls.insert(i, accessibility)
+                    end
+                  end
+                end
+              end
 
-          each_child node do |child|
-            process child, decls: decls, comments: comments, singleton: singleton
+              # For `private def foo` syntax
+              current = current_accessibility(decls)
+              decls << accessibility
+              each_child node do |child|
+                process child, decls: decls, comments: comments, context: context
+              end
+              decls << current
```

Comment:
It processes arguments of `private` and `public` with the accessibility for `private def foo` syntax.